### ISSUE TITLE
create a build for Ubuntu 20

### ISSUE THREAD

**Issue by vladzcloudius** [achaphiv/ppa-fonts#46]
Hi.
I'm really sorry to bother you again but could you, please, create a build for Ubuntu 20.04 LTS?

Just installed it and fonts rendering is still not (as) good.

Please, save us!

**Comment by achaphiv** [comment]
Yeah, I'll give it a try.

**Comment by achaphiv** [comment]
Current status: I need an updated patch for freetype 2.10.1.

I'm currently busy with work, so this might be another week or two.

**Comment by vladzcloudius** [comment]
@achaphiv No problem at all - no rush.

**Comment by vladzcloudius** [comment]
@achaphiv Sorry for bothering. I guess you are still busy.
This is just a reminder to pop in your mailbox.

**Comment by achaphiv** [comment]
As usual, debian package is convoluted.

But, the build seems to have gone through:

https://launchpad.net/~no1wantdthisname/+archive/ubuntu/ppa/+build/19316987

**Comment by vladzcloudius** [comment]
@achaphiv Hi. I guess you are still busy.
I'm willing to help this move forward. - Could you, please, tell where I can get that 2.10.1 patch? Is that something that can be taken from somewhere you are have to actually write it? If the later, can I help somehow?

Thanks in advance.

**Comment by vladzcloudius** [comment]
> As usual, debian package is convoluted.
> 
> But, the build seems to have gone through:
> 
> https://launchpad.net/~no1wantdthisname/+archive/ubuntu/ppa/+build/19316987

Thanks, man!
I noticed that the `fontconfig-infinality` is missing in the build above.
Is this expected?

**Comment by vladzcloudius** [comment]
I see that you haven't pushed them into the repo yet either. 
I guess I simply need to wait a little more... ;)

**Comment by vladzcloudius** [comment]
@achaphiv Thanks. I see the freetype appears in the web iface here: https://launchpad.net/~no1wantdthisname/+archive/ubuntu/ppa

But I still can't detect it using `dpkg`/`apt`.

**Comment by vladzcloudius** [comment]
@achaphiv QQ: I noticed slight changes in the way it operates compared to the previous version. 

Was there a change in the rendering/hinting approach? 
At first it felt as if it was "muddier" but when I looked closer it turned out that it was just my perception while rendering (I'm using Hinted Noto fonts all over the place). 
At times it also seems like there is more "shade" near vertical stems than before.

Could you, please, clarify if that's expected?

**Comment by greatquux** [comment]
currently he's only providing the freetype library with infinality patches, not any altered fontconfig files.  Generally I don't even bother with those because they make my head spin. I see that the INFINALITY environment variable name has been changed to INFINALITY_FT in the latest version of the infinality-settings.sh file (see https://github.com/pdeljanov/infinality-remix/blob/master/freetype2-infinality-remix/infinality-settings.sh) but changing it doesn't seem to have any effect for me, not sure why...

**Comment by vladzcloudius** [comment]
> currently he's only providing the freetype library with infinality patches, not any altered fontconfig files. Generally I don't even bother with those because they make my head spin. I see that the INFINALITY environment variable name has been changed to INFINALITY_FT in the latest version of the infinality-settings.sh file (see https://github.com/pdeljanov/infinality-remix/blob/master/freetype2-infinality-remix/infinality-settings.sh) but changing it doesn't seem to have any effect for me, not sure why...

@greatquux 
You are right: `fontconfig-infinality` package is missing: the only one available is `libfreetype6`.

@achaphiv Could you, please, build it for U20 too?

@greatquux I was using scripts from `fontconfig-infinality` which are slightly different from the one you referenced above.

**Comment by vladzcloudius** [comment]
@achaphiv I cleaned up the `fontconfig-infinality` (by mistake) which remained from 19.10. I backed up the [infinality-settings.sh](https://github.com/achaphiv/ppa-fonts/files/4647496/infinality-settings.zip) and after manually putting it in `/etc/profile.d/` I could clearly see a difference (compared to using "naked" `libfreetype6` without tweaking) and this is the best result I saw so far with this new build.

Namely
```
sudo /etc/fonts/infinality/infctl.sh setstyle -> choose "linux"
```
was making fonts look worse.

The only issue right now Chrome's and Slack's fonts rendering.
In particular how Chrome displays `gmail` page:

![image](https://user-images.githubusercontent.com/6103981/82270890-41dc3d80-9944-11ea-93be-b956ea91f67f.png)

Look at this ugliest spacing between letters of the same word.

Here is the same email subject as rendered by Firefox:
![image](https://user-images.githubusercontent.com/6103981/82271046-af886980-9944-11ea-9a4d-a7b7ab8607f1.png)

In the Slack letter `i` is rendered ugly:
![image](https://user-images.githubusercontent.com/6103981/82271247-47865300-9945-11ea-9034-1dfbe39f5411.png)

See how dot is has an offset to the right.

However except for that fonts look pretty sexy in general.

**Comment by greatquux** [comment]
The Chrome issue (and Slack, which uses Electron, which is Chromium-based) is because they use their own bundled freetype library: https://github.com/pdeljanov/infinality-remix/issues/3

**Comment by vladzcloudius** [comment]
> The Chrome issue (and Slack, which uses Electron, which is Chromium-based) is because they use their own bundled freetype library: [pdeljanov/infinality-remix#3](https://github.com/pdeljanov/infinality-remix/issues/3)

Thanks!
Interestingly the OSS Chromium does much better.

**Comment by achaphiv** [comment]
I stopped building `fontconfig-infinality` some time ago.

See #43.

However, I did forget to build `cairo` with the respect fontconfig patch.
That should deploy shortly.

**Comment by vladzcloudius** [comment]
> I stopped building `fontconfig-infinality` some time ago.
> 
> See #43.
> 
> However, I did forget to build `cairo` with the respect fontconfig patch.
> That should deploy shortly.

Just picked it up. Thanks, @achaphiv .
Luckily I still had `fontconfig-infinality` files on my son's laptop so I simply picked them up and installed on mine.

To your question on U17 related thread: yes, this makes a very visible difference. In particular it changes how things look when one uses small font sizes - letters become wider which makes them much more readable.
Also I see that small letters rendered much more consistently.

Don't know how but the `i` fuckup in Slack got fixed after I used `fontconfig-infinality` scripts. Somehow rendering in Chromium also seem to change - in particular the small-font-size hack mentioned above got applied onto Facebook page.

So, I personally think that it's very useful.

**Comment by achaphiv** [comment]
The `fontconfig-infinality` [deb](https://launchpad.net/~no1wantdthisname/+archive/ubuntu/ppa/+files/fontconfig-infinality_20130104-0ubuntu0ppa1_all.deb) is listed under [packages](https://launchpad.net/~no1wantdthisname/+archive/ubuntu/ppa/+packages).

As for packaging config files from: https://github.com/pdeljanov/infinality-remix/tree/master/fontconfig-infinality-remix

I can give it a try, but I doubt this will happen anytime soon.

I will (finally) be using linux again for work, so I might end up fixing it when I get my new laptop.

Staring at terrible font rendering for 8+ hours was one of the reasons why I initially started this ppa, after all.

**Comment by vladzcloudius** [comment]
> The `fontconfig-infinality` [deb](https://launchpad.net/~no1wantdthisname/+archive/ubuntu/ppa/+files/fontconfig-infinality_20130104-0ubuntu0ppa1_all.deb) is listed under [packages](https://launchpad.net/~no1wantdthisname/+archive/ubuntu/ppa/+packages).

Not for Focal... ;)

> 
> As for packaging config files from: https://github.com/pdeljanov/infinality-remix/tree/master/fontconfig-infinality-remix
> 
> I can give it a try, but I doubt this will happen anytime soon.
> 
> I will (finally) be using linux again for work, so I might end up fixing it when I get my new laptop.
> 
> Staring at terrible font rendering for 8+ hours was one of the reasons why I initially started this ppa, after all.

**Comment by vladzcloudius** [comment]
@achaphiv @greatquux Wanted to update you that after long lasting fight I figured out that the thing that apparently was bothering me the most was the 96 DPI that Gnome was enforcing while my monitor's native is 109.

As a result some letters were still "shady" and this was driving me CRAZY!!!
After I fixed the DPI I'm getting the result that I like the most with these libs without any scripting.
